superagi/tools/google_calendar/list_calendar_events.py

```diff
@@ -0,0 +1,96 @@
+class ListCalendarEventsTool(BaseTool):
+    name: str = "List Google Calendar Events"
+    args_schema: Type[BaseModel] = ListCalendarEventsInput
+    description: str = "Get the list of all the events from Google Calendar"
+    agent_id: int  = None
+    def _execute(self, start_time: str = 'None', start_date: str = 'None', end_date: str = 'None', end_time: str = 'None'):
+        engine = connect_db()
+        Session = sessionmaker(bind=engine)
+        session = Session()
+        toolkit_id = self.tool_kit_config.tool_kit_id
+        service = GoogleCalendarCreds().get_credentials(toolkit_id)
+        if service["success"]:
+            service = service["service"]
+        else:
+            return f"Kindly connect to Google Calendar"
+        date_utc = CalendarDate().get_date_utc(start_date,end_date,start_time,end_time,service)
+        event_results = (
+            service.events().list(
+            calendarId = "primary",
+            timeMin = date_utc['start_datetime_utc'],
+            timeMax = date_utc['end_datetime_utc'],
+            singleEvents = True,
+            orderBy = "startTime",
+            ).execute()
+        )
+        if not event_results:
+            return f"No events found for the given date and time range."
+        csv_data = [['Event ID','Event Name','Start Time','End Time','Attendees']]
+        for item in event_results['items']:
+            eid_url = item["htmlLink"]
+            parsed_url = urlparse(eid_url)
+            query_parameters = parse_qs(parsed_url.query)
+            event_id = query_parameters.get('eid',[None])[0]
+            summary = ""
+            start_date = ""
+            end_date = ""
+            if "summary" in item:
+                summary = item['summary']
+            if item['start'] and item['end']:
+                start_date = item['start']['dateTime']
+                end_date = item['end']['dateTime']
+            attendees = []
+            if "attendees" in item:
+                for attendee in item['attendees']:
+                    attendees.append(attendee['email'])
+            attendees_str = ','.join(attendees)
+            csv_data.append([event_id,summary,start_date,end_date,attendees_str])
+        file = datetime.now()
+        file = file.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        file_name = "Google_Calendar_" + file + ".csv"
+        final_path = file_name
+        root_dir = get_config('RESOURCES_OUTPUT_ROOT_DIR')
+        if root_dir is not None:
+            root_dir = root_dir if root_dir.startswith("/") else os.getcwd() + "/" + root_dir
+            root_dir = root_dir if root_dir.endswith("/") else root_dir + "/"
+            final_path = root_dir + file_name
+        else:
+            final_path = os.getcwd() + "/" + file_name
+        with open(final_path, "w") as file:
```

Comment:
use ResourceManager class. Add write_csv_file method in that for this use case.

Reply:
Sir checked and added s3 helper wherever required